metadata-ingestion/src/datahub/ingestion/source/sql/trino.py

```diff
@@ -175,6 +236,68 @@ def get_db_name(self, inspector: Inspector) -> str:
         else:
             return super().get_db_name(inspector)
 
+    def _get_sibling_urn(
+        self,
+        dataset_name: str,
+        inspector: Inspector,
+        schema: str,
+        table: str,
+    ) -> Optional[str]:
+        catalog_name = dataset_name.split(".")[0]
+        connector_platform_details: Optional[PlatformDetail] = None
+
+        connector_name = get_catalog_connector_name(catalog_name, inspector)
+        if connector_name:
+            connector_platform_details = KNOWN_CONNECTOR_PLATFORM_MAPPING.get(
+                connector_name
+            )
+
+        connector_details = self.config.catalog_to_connector_details.get(
+            catalog_name, ConnectorDetail()
+        )
+
+        if connector_platform_details:
+            if not connector_platform_details.is_three_tier:  # connector is two tier
+                return make_dataset_urn_with_platform_instance(
+                    platform=connector_platform_details.platform_name,
+                    name=f"{schema}.{table}",
+                    platform_instance=connector_details.platform_instance,
+                    env=connector_details.env,
+                )
+            elif connector_details.connector_database:  # connector is three tier
+                return make_dataset_urn_with_platform_instance(
+                    platform=connector_platform_details.platform_name,
+                    name=f"{connector_details.connector_database}.{schema}.{table}",
+                    platform_instance=connector_details.platform_instance,
+                    env=connector_details.env,
+                )
+
+        return None
+
+    def _process_table(
+        self,
+        dataset_name: str,
+        inspector: Inspector,
+        schema: str,
+        table: str,
+        sql_config: SQLCommonConfig,
+    ) -> Iterable[Union[SqlWorkUnit, MetadataWorkUnit]]:
+        yield from super()._process_table(
+            dataset_name, inspector, schema, table, sql_config
+        )
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+        sibling_urn = self._get_sibling_urn(dataset_name, inspector, schema, table)
```

Comment:
we should also generate lineage here. also, make sure to talk to mayuri about what needs to be done with siblings to make this work